### PR TITLE
Rescue document generation exceptions in GeoBlacklightEventGenerator

### LINKED ISSUE #5551: GeoBlacklight Event generator shouldn't build a GeoBlacklight document during DeleteMemberJob
This creates an error when the resource is a scanned map because GBL document attempts to build a IIIF manifest link, and something needed to build the manifest has already been deleted. Either the document shouldn't get built or this method should rescue an exception and return false:

https://github.com/pulibrary/figgy/blob/7b61f1dd8e0e0fff661aa17bc867c3ae3cd6a56d/app/services/event_generator/geoblacklight_event_generator.rb#L84

Figgy/production] NoMethodError: undefined method `pyramidal_derivative' for nil:NilClass

## Backtrace

    line 777 of [PROJECT_ROOT]/app/services/manifest_builder.rb: base_url
    line 756 of [PROJECT_ROOT]/app/services/manifest_builder.rb: manifest_image_path
    line 52 of [PROJECT_ROOT]/app/services/geo_discovery/document_builder/iiif_builder.rb: iiif_path

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/53391/faults/90953229)

### PR DESCRIPTION
**Pull request by eliotjordan**
Closes #5551

### CODE REVIEW

**Review by hackartisan — CHANGES_REQUESTED** [review]
I want to approve this because it clearly closes the ticket in the recommended way. But I don't really understand it and I think it's confusing that the test says "when the document is invalid it's valid".  Maybe the test can spell out the use case for this functionality more specifically? In the ticket it says "This creates an error when the resource is a scanned map because GBL document attempts to build a IIIF manifest link." (that doesn't make sense to me either; why wouldn't a scanned map have a manifest?) It also might be helpful to add a comment to `errors?` since it isn't concerned with some kinds of errors -- the kind you'd have to rescue from..

File: spec/services/event_generator/geoblacklight_event_generator_spec.rb [hunk 1]
```diff
@@ -156,5 +156,17 @@
...
     end
+
+    context "when a record generates an invalid geoblacklight document" do
+      let(:document_builder) { instance_double(GeoDiscovery::DocumentBuilder) }
+      before do
+        allow(GeoDiscovery::DocumentBuilder).to receive(:new).and_return(document_builder)
+        allow(document_builder).to receive(:to_hash).and_raise(StandardError)
+      end
+
+      it "is still valid" do
```
Comment (hackartisan) [inline spec/services/event_generator/geoblacklight_event_generator_spec.rb:167]:
```suggestion
      it "is still emits a valid event" do
```

File: app/services/event_generator/geoblacklight_event_generator.rb [hunk 2]
```diff
@@ -81,8 +81,12 @@ def slug(record)
       end
 
       def errors?(record)
-        doc = document_generator(record).to_hash
-        return true unless doc[:error].nil?
+        begin
+          doc = document_generator(record).to_hash
+          return true unless doc[:error].nil?
+        rescue
```
Comment (hackartisan) [inline app/services/event_generator/geoblacklight_event_generator.rb:87]:
```suggestion
        # the resource could not generate a document at all
        rescue
```

File: app/services/event_generator/geoblacklight_event_generator.rb [hunk 3]
```diff
@@ -81,8 +81,12 @@ def slug(record)
       end
 
       def errors?(record)
```
Comment (hackartisan) [inline app/services/event_generator/geoblacklight_event_generator.rb:83]:
```suggestion
      def schema_errors?(record)
```

**Review by hackartisan — CHANGES_REQUESTED** [review]
changes per discussion

**Comment by eliotjordan** [comment]
@hackartisan Ready for re-review.

**Pull request merged**